src/GuessControl.js

```diff
@@ -1,47 +1,71 @@
-import React, { Component } from "react";
-import Button from "./Button";
+// import React, { Component, useState } from "react";
+// import Button from "./Button";
 
-class GuessControl extends Component {
-  constructor(props) {
-    super(props);
+// class GuessControlOld extends Component {
+//   constructor(props) {
+//     super(props);
 
-    this.state = {
-      currentGuess: "",
-    };
+//     this.state = {
+//       currentGuess: "",
+//     };
 
-    /**
-     * These lines are required to make the methods/functions declared on this
-     *  class have the correct `this` object when they run.
-     */
-    this.handleInputChange = this.handleInputChange.bind(this);
-    this.onSubmitGuess = this.onSubmitGuess.bind(this);
-  }
+//     /**
+//      * These lines are required to make the methods/functions declared on this
+//      *  class have the correct `this` object when they run.
+//      */
+//     this.handleInputChange = this.handleInputChange.bind(this);
+//     this.onSubmitGuess = this.onSubmitGuess.bind(this);
+//   }
 
-  handleInputChange(event) {
-    this.setState({ currentGuess: event.target.value });
-  }
+//   handleInputChange(event) {
+//     this.setState({ currentGuess: event.target.value });
+//   }
 
-  onSubmitGuess() {
-    // Since the values from an HTML input are strings by default,
-    //  convert to a number for the returned guess value
-    //  by passing in the string to the Number function.
-    // See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number
-    this.props.onGuess(Number(this.state.currentGuess));
-    this.setState({ currentGuess: "" });
-  }
+//   onSubmitGuess() {
+//     // Since the values from an HTML input are strings by default,
+//     //  convert to a number for the returned guess value
+//     //  by passing in the string to the Number function.
+//     // See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number
+//     this.props.onGuess(Number(this.state.currentGuess));
+//     this.setState({ currentGuess: "" });
+//   }
 
-  render() {
-    return (
-      <div>
-        <input
-          type="number"
-          value={this.state.currentGuess}
-          onChange={this.handleInputChange}
-        />
-        <Button onClick={this.onSubmitGuess}>Submit Guess</Button>
-      </div>
-    );
-  }
-}
+//   render() {
+//     return (
+//       <div>
+//         <input
+//           type="number"
+//           value={this.state.currentGuess}
+//           onChange={this.handleInputChange}
+//         />
+//         <Button onClick={this.onSubmitGuess}>Submit Guess</Button>
+//       </div>
+//     );
+//   }
+// }
 
-export default GuessControl;
+// export default GuessControlOld;
+import React, { useState } from "react";
+import Button from "./Button";
+
+export default function GuessControl(data) {
+  const [currentGuess, setCurrentGuess] = useState("")
```

Comment:
did you understand that withRouter does not subscribe to location changes like React Redux’s connect does for state changes. Instead, re-renders after location changes propagate out from the <Router> component which means that withRouter does not re-render on route transitions unless its parent component re-renders?

Reply:
I did not. I am researching withRouter which I have not used in react router, the connect function and more about their differences.

Reply:
it looks like withRouter has depreciated with v6 but you can recreate it as a custom hook: 
import { useHistory } from 'react-router-dom';

export const withRouter = (Component) => {
  const Wrapper = (props) => {
    const history = useHistory();
    
    return (
      <Component
        history={history}
        {...props}
        />
    );
  };
  
  return Wrapper;
};

Reply:
i have not yet used redux in my projects but would like to incorporate it in my next project. after reading and watching about connect it looks very handy. i think it would be fun to refactor this with another branch that will use connect and/ or withRouter. thank you.